analyses/survival-analysis/02-summary-statics-TP53.Rmd

```diff
@@ -0,0 +1,268 @@
+}
+
+ggplot(combined, aes(x = age_at_diagnosis_days, y = tp53_score)) + 
+  geom_smooth(method = "lm") + 
+  geom_point() + 
+  geom_text(y = 0.05, x = 20000, 
+            label = lm_eqn(combined, x= combined$age_at_diagnosis_days, y = combined$tp53_score), 
+            parse = TRUE)
+```
+
+## The distribution of TP53 score between age range and tp53 score 
+
+```{r}
+group_summary <- combined %>%
+  group_by(binned_age) %>%
+  summarize(count = n())
+
+# Filter the groups with at least 3 samples
+combined_filtered <- combined %>%
+  filter(binned_age %in% group_summary$binned_age[group_summary$count >= 3])
+
+ggplot(combined_filtered, aes(x = factor(binned_age, levels = c("[0,15]", "(15,26]", "(26,40]", "(40,62]", ">62", "not reported")), y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(aes(color = broad_mol), width = 0.3, alpha = 0.5) + 
+  stat_compare_means(comparisons = list(c("[0,15]", "(15,26]"), c("[0,15]", "(26,40]"), c("[0,15]", "(40,62]"), c("[0,15]", ">62"))) + 
+  xlab("") + 
+  theme_Publication()
+
+```
+## save the plot
+
+```{r}
+ggsave(file.path(plots_dir, "boxplot_tp53_score_binned_age.png"))
+
+```
+
+```{r}
+p1 <- ggplot(combined, aes(x = reported_gender, y = tp53_score)) + 
+      geom_boxplot() +
+      geom_jitter(aes(color = broad_mol), width = 0.3, alpha = 0.5) + 
+      stat_compare_means(comparisons = list(c("Female", "Male"))) + 
+      ggtitle(label = "All samples") + 
+      xlab("") + 
+      theme_Publication()
+
+p2 <- ggplot(HGG_survival, aes(x = reported_gender, y = tp53_score)) + 
+      geom_boxplot() +
+      geom_jitter(width = 0.3, alpha = 0.5) + 
+      stat_compare_means(comparisons = list(c("Female", "Male"))) + 
+      ggtitle(label = "HOPE cohort") + 
+      xlab("") + 
+      theme_Publication()
+
+p3 <- ggplot(GBM_df, aes(x = reported_gender, y = tp53_score)) + 
+      geom_boxplot() +
+      geom_jitter(width = 0.3, alpha = 0.5) + 
+      stat_compare_means(comparisons = list(c("Female", "Male"))) + 
+      ggtitle(label = "GBM") + 
+      xlab("") + 
+      theme_Publication()
+
+p1 + p2 + p3
+
+```
+
+## save the plot
+
+```{r}
+ggsave(file.path(plots_dir, "boxplot_tp53_score_gender.png"), height = 6, width = 12)
+
+```
+
+
+## The distribution of TP53 score among different races 
+
+```{r}
+group_summary <- HGG_survival %>%
+  group_by(race) %>%
+  summarize(count = n())
+
+# Filter the groups with at least 3 samples
+HGG_filtered <- HGG_survival %>%
+  filter(race %in% group_summary$race[group_summary$count >= 3])
+
+p1 <- HGG_filtered %>% 
+  mutate(race = fct_reorder(race, -tp53_score, .fun='median' )) %>%
+  ggplot(aes(x = race, y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(width = 0.3, alpha = 0.5) + 
+  xlab("") +
+  stat_compare_means(comparisons = list(c("Asian", "Reported Unknown"), c("Asian", "Black or African American"), c("Asian", "White"), c("Asian", "Other"))) +
+  theme_Publication() + 
+  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
+
+p1
```

Comment:
I can check how many of these patients currently have ancestry prediction by Somalier...it would be interesting to see the proportions with `reported_race == "Asian"` that fall into East Asian and South Asian ancestries, and if there is any association with TP53 score